Pi+/Coxeter/MCoxeter.agda

```diff
@@ -4,134 +4,28 @@ module Pi+.Coxeter.MCoxeter where
-long k l r = ext (long≅ k l r _ _ idp idp)
-
-braid : {n : ℕ} -> (l r : List) -> (l ++ S n :: n :: S n :: r) ≅* (l ++ n :: S n :: n :: r)
-braid {n} l r = long {n} 0 l r
-
-trans : {m1 m2 m3 : List} -> (m1 ≅* m2) -> (m2 ≅* m3) -> m1 ≅* m3
-trans idp p  = p
-trans (trans≅ x q) p = trans≅ x (trans q p)
-
-l++≅ : (m1 m2 l : List) -> (m1 ≅ m2) -> ((l ++ m1) ≅ (l ++ m2))
-l++≅ m1 m2 l (cancel≅ l₁ r .m1 .m2 defm defmf) =  cancel≅ (l ++ l₁) r _ _ (≡-trans (start+end idp defm) (≡-sym (++-assoc l l₁ _))) ((≡-trans (start+end idp defmf) (≡-sym (++-assoc l l₁ _))))
-l++≅ m1 m2 l (swap≅ x l₁ r .m1 .m2 defm defmf) =  swap≅ x (l ++ l₁) r _ _ (≡-trans (start+end idp defm) (≡-sym (++-assoc l l₁ _))) ((≡-trans (start+end idp defmf) (≡-sym (++-assoc l l₁ _))))
-l++≅ m1 m2 l (long≅ k l₁ r .m1 .m2 defm defmf) =  long≅ k (l ++ l₁) r _ _ (≡-trans (start+end idp defm) (≡-sym (++-assoc l l₁ _))) ((≡-trans (start+end idp defmf) (≡-sym (++-assoc l l₁ _))))
-
-l++ : (l : List) -> {m1 m2 : List} -> (m1 ≅* m2) -> ((l ++ m1) ≅* (l ++ m2))
-l++ l idp = idp
-l++ l (trans≅ x p) = trans≅ (l++≅ _ _ l x) ((l++ l p))
-
-++r≅ : (m1 m2 r : List) -> (m1 ≅ m2) -> ((m1 ++ r) ≅ (m2 ++ r))
-++r≅ m1 m2 r (cancel≅ l r₁ .m1 .m2 defm defmf) = cancel≅ l (r₁ ++ r)  _ _  (≡-trans (start+end defm idp) (++-assoc l _ r)) ((≡-trans (start+end defmf idp) (++-assoc l _ r)))
-++r≅ m1 m2 r (swap≅ x l r₁ .m1 .m2 defm defmf) = swap≅ x l (r₁ ++ r)  _ _  (≡-trans (start+end defm idp) (++-assoc l _ r)) ((≡-trans (start+end defmf idp) (++-assoc l _ r)))
-++r≅ m1 m2 r (long≅ k l r₁ .m1 .m2 defm defmf) = long≅ k l (r₁ ++ r)  _ _
-  (≡-trans (start+end defm idp) (≡-trans (++-assoc l _ r) (start+end (idp {a = l}) (head+tail idp (head+tail idp (++-assoc (_ ↓ k) _ r))))))
-  ((≡-trans (start+end defmf idp) (≡-trans (++-assoc l _ r) (start+end (idp {a = l}) (head+tail idp (head+tail idp (head+tail idp (++-assoc _ r₁ r))))))))
-
-++r : {m1 m2 : List} -> (r : List) -> (m1 ≅* m2) -> ((m1 ++ r) ≅* (m2 ++ r))
-++r r idp = idp
-++r r (trans≅ x p) = trans≅ (++r≅ _ _ r x) (++r r p)
-
-idp≅* : {l l' : List} -> (l == l') -> l ≅* l'
-idp≅* idp = idp
-
-module ≅*-Reasoning where
-    infix  1 ≅*begin_
-    infixr 2 _≅*⟨⟩_ _≅*⟨_⟩_
-    infix  3 _≅*∎
-
-    ≅*begin_ : ∀ {x y : List}
-             → x ≅* y
-               -----
-             → x ≅* y
-    ≅*begin x≅*y  =  x≅*y
-
-    _≅*⟨⟩_ : ∀ (x : List) {y : List}
-            → x ≅* y
-              -----
-            → x ≅* y
-    x ≅*⟨⟩ x≅*y  =  x≅*y
-
-    _≅*⟨_⟩_ : ∀ (x : List) {y z : List}
-             → x ≅* y
-             → y ≅* z
-               -----
-             → x ≅* z
-    x ≅*⟨ x≅*y ⟩ y≅*z  = trans x≅*y y≅*z
+open ≅*-Reasoning
 
-    _≅*∎ : ∀ (x : List)
-           -----
-          → x ≅* x
-    x ≅*∎  =  idp
 
-open ≅*-Reasoning
+data _≃_ : List -> List -> Set where
+  comm≅ : {m1 m2 mf : List} -> (p1 : m1 ≅* mf) -> (p2 : m2 ≅* mf) -> m1 ≃ m2
```

Comment:
Can you write it like this instead? `m1 ≃ m2 = Σ _ λ mf → m1 ≅* mf × m2 ≅* mf`, this is more idiomatic and easier to work with when we truncate.